homeassistant/components/schlage/select.py

```diff
@@ -0,0 +1,86 @@
+"""Platform for Schlage select integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SchlageDataUpdateCoordinator
+from .entity import SchlageEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class SchlageSelectEntityDescription(SelectEntityDescription):
+    """Entity description for a Schlage select."""
+
+
+_SECONDS_TO_OPTIONS = {
+    0: "disabled",
+    15: "15 seconds",
+    30: "30 seconds",
+    60: "60 seconds",
+    120: "120 seconds",
+    240: "240 seconds",
+    300: "300 seconds",
+}
+_OPTIONS_TO_SECONDS = {d: s for s, d in _SECONDS_TO_OPTIONS.items()}
+
+
+_OPTIONS: tuple[SchlageSelectEntityDescription] = (
+    SchlageSelectEntityDescription(
+        key="auto_lock_time",
+        translation_key="auto_lock_time",
+        entity_category=EntityCategory.CONFIG,
+        options=list(_OPTIONS_TO_SECONDS),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up selects based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
```

Comment:
Make sure this supports auto-removal of the entry. See https://github.com/dknowles2/ha-core/blob/dev/homeassistant/components/schlage/binary_sensor.py#L49-L61 for how this is done with other entities.

Reply:
Sure